perllib/FixMyStreet/DB/ResultSet/DefectType.pm

```diff
@@ -3,20 +3,38 @@ use base 'DBIx::Class::ResultSet';
 
 use strict;
 use warnings;
+use JSON::MaybeXS;
 
 sub for_bodies {
     my ($rs, $bodies, $category) = @_;
     my $attrs = {
         'me.body_id' => $bodies,
     };
-    if ($category) {
-        $attrs->{'contact.category'} = [ $category, undef ];
-    }
-    $rs->search($attrs, {
+    my $filters = {
         order_by => 'name',
         join => { 'contact_defect_types' => 'contact' },
         distinct => 1,
-    });
+    };
+    if ($category) {
+        $attrs->{'contact.category'} = [ $category, undef ];
+    }
+    $rs->search($attrs, $filters);
+}
+
+sub by_categories {
+    my ($rs, $area_id, @contacts) = @_;
+    my $body_id = FixMyStreet::DB->resultset('BodyArea')->find({ area_id => $area_id })->body_id;
```

Comment:
Theoretically, an area could cover multiple bodies so it would be good to cover this and have it fetch all the matching bodies.